sympy/stats/tests/test_discrete_rv.py

```diff
@@ -172,12 +213,13 @@ def test_conditional():
     assert P(X < 2, Eq(X, 2)) == 0
     assert P(X > 2, Eq(X, 3)) == 1
 
+
 def test_product_spaces():
     X1 = Geometric('X1', S(1)/2)
     X2 = Geometric('X2', S(1)/3)
     assert str(P(X1 + X2 < 3, evaluate=False)) == """Sum(Piecewise((2**(X2 - n - 2)*(2/3)**(X2 - 1)/6, """\
```

Comment:
This could just be:
 ```py
assert P(X1 + X2 < 3, evaluate=False).dummy_eq(Sum(Piecewise(
    (2**(1 - X2)*2**(X2 - n - 2)/4, (-X2 + n + 3 >= 1) & 
    (-X2 + n + 3 < oo)), (0, True)), (X2, 1, oo), (n, -oo, -1)))
```

Reply:
This is giving `false` as the answer. Could you please guide me on how I can make it work.

Reply:
It's giving False because the expression there is not the same as the one returned (or the one already in the diff here).

The point is not that you should copy the exact line of code here but that you should use dummy_eq rather than string comparison.

Reply:
@oscarbenjamin I tried multiple ways, but not getting `True` for `dummy_eq`. Is there some way I could get this?

Reply:
I just tried this to see what the expression is and I get
```julia
In [10]: from sympy.stats import *                                                                                                

In [11]:     X1 = Geometric('X1', S(1)/2) 
    ...:     X2 = Geometric('X2', S(1)/3)                                                                                         

In [12]: res = P(X1 + X2 < 3, evaluate=False)                                                                                     

In [13]: res                                                                                                                      
Out[13]: 
  -1     ∞                                                                 
______ ______                                                              
╲      ╲                                                                   
 ╲      ╲                                                                  
  ╲      ╲    ⎧ X₂ - n - 2    X₂ - 1                                       
   ╲      ╲   ⎪2          ⋅2/3                                             
    ╲      ╲  ⎪─────────────────────  for -X₂ + n + 3 ≥ 1 ∧ -X₂ + n + 3 < ∞
    ╱      ╱  ⎨          6                                                 
   ╱      ╱   ⎪                                                            
  ╱      ╱    ⎪          0                          otherwise              
 ╱      ╱     ⎩                                                            
╱      ╱                                                                   
‾‾‾‾‾‾ ‾‾‾‾‾‾                                                              
False  X₂ = 1
```
which looks like gibberish.

Why is the outer sum going from False to -1 and what does that even mean?
```
In [14]: repr(res)                                                                                                                
Out[14]: 'Sum(Piecewise((2**(X2 - n - 2)*(2/3)**(X2 - 1)/6, (-X2 + n + 3 >= 1) & (-X2 + n + 3 < oo)), (0, True)), (X2, 1, oo), (n, -oo, -1))'
```
You'll need to make sure to change 2/3 to S(2)/3 if using the repr above:
```
In [63]: expected = Sum(Piecewise((2**(X2 - n - 2)*(S(2)/3)**(X2 - 1)/6, (-X2 + n + 3 >= 1) & (-X2 + n + 3 < oo)), (0, True)), (X2
    ...: , 1, oo), (n, -oo, -1))
```

The reason dummy_eq doesn't work is maybe because X2 isn't a Dummy symbol. In fact in the two expressions you are comparing one has X2 as a Symbol and the other as a RandomSymbol.
```
In [62]: res.replace(X2.symbol, Dummy(integer=True)).dummy_eq(expected.replace(X2, Dummy(integer=True)))                          
Out[62]: True
```
(integer=True is needed to eliminate the < oo clause.)

I'm not sure that the output you have here is sensible though...

Reply:
@oscarbenjamin thank you for the insights. Actually this is not my code. I thought of doing some refactoring along with the my code. I will see the result and open another PR for this.

Reply:
@sidhantnagpal All other reviews have been resolved. Only this one is left. I will open another PR for this. Please consider merging if rest looks good.